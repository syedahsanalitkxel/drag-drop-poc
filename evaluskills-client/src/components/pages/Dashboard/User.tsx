import React, { useState } from 'react';
import { Button, Modal, ModalBody, ModalFooter, ModalHeader } from 'reactstrap';
import { RouteComponentProps, withRouter } from 'react-router-dom';
import User from "../../../interfaces/User";
import DashboardTemplate from '../../templates/DashboardTemplate';
import PageBody from '../../atoms/PageBody';
import PageHeader from '../../atoms/PageHeader';
import Pager from '../../molecules/Pager';

const usersData = [
    { id: 1, name: 'Robby Rash', role: 'Admin', email: 'robbyrash@gmail.com' },
    { id: 2, name: 'Jhon Doe', role: 'User', email: 'jhondoe@gmail.com' },
    { id: 3, name: 'Bella William', role: 'Admin', email: 'bellawilliam@gmail.com' },
    { id: 4, name: 'Rock Rash', role: 'User', email: 'rockrash@gmail.com' }
];

const DashboardHome: React.FunctionComponent<RouteComponentProps> = ({ history }) => {
    const [modalVisible, setModalVisible] = useState(false);

    const toggleFilterModal = () => {
        setModalVisible(!modalVisible);
    }

    const filterAction = (event: React.MouseEvent) => {
        // alert(`Filter button clicked ${event.timeStamp}`);
        event.preventDefault();
        toggleFilterModal()
    };

    const searchHandler = (searchQuery: string) => {
        alert(searchQuery);
    };

    const addAction = (event: React.MouseEvent) => {
        history.push('/users/add');
        // alert(`add button clicked ${event.timeStamp}`);
    };

    const renderUserData = (userData: User) => {
        return (
            <React.Fragment key = {userData.id}>
                <tr>
                    <td>{userData.id}</td>
                    <td>
                        <strong>{userData.name}</strong>
                    </td>
                    <td>
                        <strong>{userData.role}</strong>
                    </td>
                    <td>{userData.email}</td>
                </tr>
            </React.Fragment>
        )
    };

    return (
        <DashboardTemplate>
            <div className="row">
                <div className="col-lg-12">
                    <PageHeader
                        title="Users"
                        filterAction={filterAction}
                        searchHandler={searchHandler}
                        actionButtonText="Add User"
                        actionHandler={addAction}
                    />
                    <PageBody>
                        <div className="ibox m-b-15">
                            <div className="ibox-content">
                                <table className="table">
                                    <thead>
                                    <tr>
                                        <th>#</th>
                                        <th>Name</th>
                                        <th>Role</th>
                                        <th>Email</th>
                                    </tr>
                                    </thead>
                                    <tbody>
                                    {usersData.map(renderUserData)}
                                    </tbody>
                                </table>
                            </div>
                        </div>
                        <Pager />
                    </PageBody>
                </div>
            </div>


            {/* Filter Modal */}
            <div>
                <Modal isOpen={modalVisible} toggle={toggleFilterModal}>
                    <ModalHeader toggle={toggleFilterModal}>
                        Filters
                    </ModalHeader>
                    <ModalBody>
                        <div className="form-group row">
                            <label className="col-sm-4 col-form-label font-bold">Role</label>
                            <div className="col-sm-8">
                                <select className="form-control m-b col-sm-12" name="role" >
                                    <option>Select Role</option>
                                    <option>option 2</option>
                                    <option>option 3</option>
                                    <option>option 4</option>
                                </select>
                            </div>
                        </div>
                        <div className="hr-line-dashed" />
                        <div className="form-group row">
                            <label className="col-sm-4 col-form-label font-bold">Clients</label>
                            <div className="col-sm-8">
                                <select className="form-control m-b col-sm-12" name="name" >
                                    <option>Select Clients</option>
                                    <option>option 2</option>
                                    <option>option 3</option>
                                    <option>option 4</option>
                                </select>
                            </div>
                        </div>
                    </ModalBody>
                    <ModalFooter>
                        <Button color="primary" onClick={toggleFilterModal}>Reset</Button>{' '}
                        <Button color="secondary" onClick={toggleFilterModal}>Apply</Button>
                    </ModalFooter>
                </Modal>
            </div>

        </DashboardTemplate>
    );
};

export default withRouter(DashboardHome);