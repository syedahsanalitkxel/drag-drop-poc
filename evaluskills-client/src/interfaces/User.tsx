export default interface User {
  name: string;
  role: string;
  email: string;
  id: number;
}
