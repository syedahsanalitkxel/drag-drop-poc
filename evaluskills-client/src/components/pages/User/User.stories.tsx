import React from 'react';

import { storiesOf } from '@storybook/react';

import User from '.';
import UserList from '../../../interfaces/UserList';
import UsersFilterInterface from '../../../interfaces/UserFilter';

const usersData: any = [];
function filterHandler(filters: UsersFilterInterface) {
  // fetchAllUsers({ ...state.filters, ...filters });
}

function submitForm(values: any, action: string, id?: string) {
  // fetchAllUsers({ ...state.filters, ...filters });
}
storiesOf('Dashboard', module).add('User', () => (
  <User Users={usersData} filterHandler={filterHandler} submitForm={submitForm}>
    User Content
  </User>
));
