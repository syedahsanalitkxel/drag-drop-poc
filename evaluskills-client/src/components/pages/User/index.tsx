import React, { useContext, useEffect, useState } from 'react';
import { Button, Modal, ModalBody, ModalFooter, ModalHeader } from 'reactstrap';
import AddUser from '../../pages/AddUser';
import User from '../../../interfaces/User';
import DashboardTemplate from '../../templates/DashboardTemplate';
import PageBody from '../../atoms/PageBody';
import PageHeader from '../../atoms/PageHeader';
import Pager from '../../molecules/Pager';
import UsersList from '../../organisms/UserList';
import Editcomponent from '../../pages/AddUser';

interface Props {
  changeListener?: (formValues: User) => void;
}

const initialState = {
  email: ' ',
  firstName: ' ',
  id: ' ',
  lastName: ' ',
  role: ' ',
};

const usersData = [
  { id: '1', firstName: 'Robby', lastName: 'Rash', role: 'Admin', email: 'robbyrash@gmail.com' },
  { id: '2', firstName: 'Jhon', lastName: 'Doe', role: 'User', email: 'jhondoe@gmail.com' },
  {
    id: '3',
    firstName: 'Bella',
    lastName: 'William',
    role: 'Admin',
    email: 'bellawilliam@gmail.com',
  },
  { id: '4', firstName: 'Rash', lastName: 'Rash', role: 'User', email: 'rockrash@gmail.com' },
];

const DashboardHome: React.FunctionComponent<Props> = ({ changeListener }) => {
  const [modalVisible, setModalVisible] = useState(false);
  const [addUserModalVisible, setAddUserModalVisible] = useState(false);
  const [editUserModalVisible, setEditUserModalVisible] = useState(false);
  const [formState, setFormState] = useState(initialState);

  useEffect(() => {
    if (changeListener) {
      changeListener(formState);
    }
  });

  const toggleEditUserModal = () => {
    setEditUserModalVisible(!editUserModalVisible);
  };

  const toggleFilterModal = () => {
    setModalVisible(!modalVisible);
  };

  const toggleAddUserModal = () => {
    setAddUserModalVisible(!addUserModalVisible);
  };

  const filterAction = (event: React.MouseEvent) => {
    // alert(`Filter button clicked ${event.timeStamp}`);
    event.preventDefault();
    toggleFilterModal();
  };

  const searchHandler = (searchQuery: string) => {
    // alert(searchQuery);
  };

  const addAction = (event: React.MouseEvent) => {
    event.preventDefault();
    setFormState(initialState);
    toggleAddUserModal();
    // history.push('/users/add');
    // alert(`add button clicked ${event.timeStamp}`);
  };

  const editAction = (userId: string) => {
    // event.preventDefault();
    const userD: any = usersData.find(user => user.id === userId);
    setFormState(userD);
    toggleEditUserModal();
    // history.push(`/users/edit/${userId}`);
  };

  const removeAction = (userId: string) => {
    // history.push(`/users/remove/${userId}`);
  };

  return (
    <DashboardTemplate>
      <div className="row">
        <div className="col-lg-12">
          <PageHeader
            title="Users"
            filterAction={filterAction}
            searchHandler={searchHandler}
            actionButtonText="Add User"
            actionHandler={addAction}
          />
          <PageBody>
            <div className="ibox m-b-15">
              <div className="table-holder">
                <UsersList listData={usersData} edit={editAction} remove={removeAction} />
              </div>
            </div>
            <Pager />
          </PageBody>
        </div>
      </div>
      {/* Filter Modal */}
      <div>
        <Modal isOpen={modalVisible} toggle={toggleFilterModal}>
          <ModalHeader toggle={toggleFilterModal}>Filters</ModalHeader>
          <ModalBody>
            <div className="form-group row">
              <label className="col-sm-4 col-form-label font-bold">Role</label>
              <div className="col-sm-8">
                <select className="form-control m-b col-sm-12" name="role">
                  <option>Select Role</option>
                  <option>option 2</option>
                  <option>option 3</option>
                  <option>option 4</option>
                </select>
              </div>
            </div>
            <div className="hr-line-dashed" />
            <div className="form-group row">
              <label className="col-sm-4 col-form-label font-bold">Clients</label>
              <div className="col-sm-8">
                <select className="form-control m-b col-sm-12" name="name">
                  <option>Select Clients</option>
                  <option>option 2</option>
                  <option>option 3</option>
                  <option>option 4</option>
                </select>
              </div>
            </div>
          </ModalBody>
          <ModalFooter>
            <Button color="primary" onClick={toggleFilterModal}>
              Reset
            </Button>
            <Button color="secondary" onClick={toggleFilterModal}>
              Apply
            </Button>
          </ModalFooter>
        </Modal>
      </div>
      <AddUser
        visible={addUserModalVisible}
        toggle={toggleAddUserModal}
        name="Add"
        FormValues={formState}
      />
      <Editcomponent
        visible={editUserModalVisible}
        toggle={toggleEditUserModal}
        name="Edit"
        FormValues={formState}
      />
      ;
    </DashboardTemplate>
  );
};

export default DashboardHome;
