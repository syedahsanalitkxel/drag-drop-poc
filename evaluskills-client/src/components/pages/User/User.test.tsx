import { configure, mount, render, shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import User from '.';
import ReactDOM from 'react-dom';

configure({ adapter: new Adapter() });
describe('User', () => {
  const user = shallow(<User />);

  it('render successfully', () => {
    expect(user).toMatchSnapshot();
  });

  it('check table is defined or not', () => {
    expect(user.find('.table')).toBeDefined();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router>
        <User />
      </Router>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('to be defind of form control `.table', () => {
    const wrapper = shallow(<User />);
    expect(wrapper.find('.table')).toBeDefined();
  });

  it('5 Rows form `.table', () => {
    const wrapper = mount(
      <Router>
        <User />
      </Router>
    );
    expect(wrapper.find('tr')).toHaveLength(5);
    expect(wrapper.find('.table')).toBeDefined();
  });
});
