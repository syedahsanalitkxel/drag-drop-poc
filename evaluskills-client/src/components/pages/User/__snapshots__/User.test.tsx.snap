// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User render successfully 1`] = `
<DashboardTemplate>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <PageHeader
        actionButtonText="Add User"
        actionHandler={[Function]}
        filterAction={[Function]}
        searchHandler={[Function]}
        title="Users"
      />
      <PageBody
        card={false}
      >
        <div
          className="ibox m-b-15"
        >
          <div
            className="ibox-content"
          >
            <UsersList
              edit={[Function]}
              listData={
                Array [
                  Object {
                    "email": "robbyrash@gmail.com",
                    "firstName": "Robby",
                    "id": "1",
                    "lastName": "Rash",
                    "role": "Admin",
                  },
                  Object {
                    "email": "jhondoe@gmail.com",
                    "firstName": "Jhon",
                    "id": "2",
                    "lastName": "Doe",
                    "role": "User",
                  },
                  Object {
                    "email": "bellawilliam@gmail.com",
                    "firstName": "Bella",
                    "id": "3",
                    "lastName": "William",
                    "role": "Admin",
                  },
                  Object {
                    "email": "rockrash@gmail.com",
                    "firstName": "Rash",
                    "id": "4",
                    "lastName": "Rash",
                    "role": "User",
                  },
                ]
              }
              remove={[Function]}
            />
          </div>
        </div>
        <Pager />
      </PageBody>
    </div>
  </div>
  <div>
    <Modal
      autoFocus={true}
      backdrop={true}
      backdropTransition={
        Object {
          "mountOnEnter": true,
          "timeout": 150,
        }
      }
      centered={false}
      fade={true}
      isOpen={false}
      keyboard={true}
      modalTransition={
        Object {
          "timeout": 300,
        }
      }
      onClosed={[Function]}
      onOpened={[Function]}
      role="dialog"
      toggle={[Function]}
      zIndex={1050}
    >
      <ModalHeader
        charCode={215}
        closeAriaLabel="Close"
        tag="h5"
        toggle={[Function]}
        wrapTag="div"
      >
        Filters
      </ModalHeader>
      <ModalBody
        tag="div"
      >
        <div
          className="form-group row"
        >
          <label
            className="col-sm-4 col-form-label font-bold"
          >
            Role
          </label>
          <div
            className="col-sm-8"
          >
            <select
              className="form-control m-b col-sm-12"
              name="role"
            >
              <option>
                Select Role
              </option>
              <option>
                option 2
              </option>
              <option>
                option 3
              </option>
              <option>
                option 4
              </option>
            </select>
          </div>
        </div>
        <div
          className="hr-line-dashed"
        />
        <div
          className="form-group row"
        >
          <label
            className="col-sm-4 col-form-label font-bold"
          >
            Clients
          </label>
          <div
            className="col-sm-8"
          >
            <select
              className="form-control m-b col-sm-12"
              name="name"
            >
              <option>
                Select Clients
              </option>
              <option>
                option 2
              </option>
              <option>
                option 3
              </option>
              <option>
                option 4
              </option>
            </select>
          </div>
        </div>
      </ModalBody>
      <ModalFooter
        tag="div"
      >
        <Button
          color="primary"
          onClick={[Function]}
          tag="button"
        >
          Reset
        </Button>
        <Button
          color="secondary"
          onClick={[Function]}
          tag="button"
        >
          Apply
        </Button>
      </ModalFooter>
    </Modal>
  </div>
  <AddUser
    FormValues={
      Object {
        "email": " ",
        "firstName": " ",
        "id": " ",
        "lastName": " ",
        "role": " ",
      }
    }
    name="Add"
    toggle={[Function]}
    visible={false}
  />
  <AddUser
    FormValues={
      Object {
        "email": " ",
        "firstName": " ",
        "id": " ",
        "lastName": " ",
        "role": " ",
      }
    }
    name="Edit"
    toggle={[Function]}
    visible={false}
  />
  ;
</DashboardTemplate>
`;
